packages/react-router/src/transformer.ts

```diff
@@ -130,21 +130,19 @@ const transformers = [
     // Key
     'formData',
     // Check
-    (v) => v instanceof FormData,
+    (v): v is FormData => v instanceof FormData,
     // To
-    (v: FormData) => {
-      const entries: Record<string, any> = {}
-      v.forEach((value, key) => {
-        entries[key] = value
-      })
+    (v) => {
+      const entries: Record<string, Array<any>> = {}
+      v.forEach((value, key) => (entries[key] ??= []).push(value))
```

Comment:
Is there anything that can be done here where all keys aren't by default an array? I'd rather switch it to an array **only** if multiple instances of the same `name` are encountered.

If you've got better reasoning for sticking with this approach, I'm open to pushback on this one.

Reply:
just because it simplifies the logic, and it's only two extra characters when serialised. I can add the logic to switch it so you can see the difference

Reply:
updated